### ISSUE TITLE
IPSec Tunnel Management Doesn't allow to specify phase details

### ISSUE THREAD

**Issue by ashwatthb** [oracle/terraform-provider-oci#1692]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### New or Affected Resource(s)

oci_core_ipsec_connection_tunnel_management

### Potential Terraform Configuration

```hcl
resource "oci_core_ipsec_connection_tunnels" "test_ip_sec_connection_tunnels" {
  phase_one_details {
  custom_authentication_algorithm = ""
  custom_dh_group = ""
  custom_encryption_algorithm = ""
  is_custom_phase_one_config = ""
  is_ike_established = ""
  lifetime = ""
  negotiated_authentication_algorithm = ""
  negotiated_dh_group = ""
  negotiated_encryption_algorithm = ""
  remaining_lifetime = ""
  remaining_lifetime_last_retrieved = ""
 }
}
```

**Comment by realolap** [comment]
Hi, I faced this issue aswell when connecting to a VPN which does not support the default DH group 5.
This limitation on the provider means that I have to configure a central part of our infrastructure outside of Terraform.

The resource provider should allow setting the custom configuration:

  phase_one_details {
    custom_authentication_algorithm = ""
    custom_dh_group                 = ""
    custom_encryption_algorithm     = ""
    is_custom_phase_one_config      = true
    lifetime                        = 28800
  }

  phase_two_details {
    custom_encryption_algorithm     = ""
    custom_authentication_algorithm = ""
    dh_group                        = ""
    is_custom_phase_two_config      = true
    is_esp_established              = true/false
    is_pfs_enabled                  = true/false
    lifetime                        = 3600
  }

**Comment by iejones** [comment]
A bug fix to set phase 1 and 2 tunnel details will be released on November 15th, 2023

**Comment by iejones** [comment]
This fix was released with version [v5.21.0](https://github.com/oracle/terraform-provider-oci/releases/tag/v5.21.0). See phase_one_details and phase_one_details in the [documentation](https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/core_ipsec_connection_tunnel_management) and the [example](https://github.com/oracle/terraform-provider-oci/blob/v5.21.0/examples/networking/ipsec_connections/ipsec_connection.tf#L127).
